tests/framework/builder.py

```diff
@@ -4,29 +4,45 @@
 
 import json
 import os
+import tempfile
+from pathlib import Path
 from conftest import init_microvm
-from framework.artifacts import Artifact, DiskArtifact
+from framework.artifacts import Artifact, DiskArtifact, Snapshot, SnapshotType
+import host_tools.logging as log_tools
+import framework.utils as utils
 
 
 class MicrovmBuilder:
     """Build fresh microvms or restore from snapshot."""
 
-    def __init__(self, root_path, bin_cloner_path):
+    def __init__(self, bin_cloner_path):
         """Initialize microvm root and cloning binary."""
-        self.root_path = root_path
         self.bin_cloner_path = bin_cloner_path
+        self.init_root_path()
 
-    def build(self, kernel: Artifact, disks: [DiskArtifact], config: Artifact):
+    def init_root_path(self):
+        """Initialize microvm root path."""
+        self.root_path = tempfile.mkdtemp(prefix="fctest-")
```

Comment:
nit: move "fctest" to a constant

Reply:
Prefixing is a good idea as it's useful for tracking down when something breaks down, but it should contain suggestive prefixes - such as all root paths created within a test, should have the same prefix.

Reply:
Added a constant for this.